### DISCUSSION TITLE
Code linter for Objeck?

### DISCUSSION THREAD

**Discussion by ghost** [objeck/objeck-lang#381] [General]
### Discussed in https://github.com/objeck/objeck-lang/discussions/164

<div type='discussions-op-text'>

<sup>Originally posted by **ghost** February 27, 2023</sup>
Something like AStyle for C/C++? @objeck </div>

The original thread is only about code formatter. What I'm suggesting is a code linter. Objeck supports many alternative syntaxes. For example, with or without the semicolon at the end of each statement. The official Objeck code written by you, @objeck, seems to default to the original syntax, which is to always have a semicolon at the end of each statement. Code written by other people isn't necessary to follow this convention. If they want to upstream their code to you, the problem of inconsistency in syntax will arise. There should be a tool to convert back and forth between the syntaxes.

**Comment by objeck** [comment]
Objeck has a source code formatter. It is part of the LSP codebase, however, it could be migrated to be a general-purpose library.

**Reply by ghost** [reply]
Please re-read my post again. Please pay attention to what I wrote about the semicolons.
